integration-tests/test_selinux.py

```diff
@@ -0,0 +1,287 @@
+                pass
+
+
+# Classes
+class SELinuxDenialsChecker:
+    """Context manager for checking SELinux denials during a time period.
+    This context manager automatically tracks start_time and end_time,
+    removing the need for manual time tracking in tests.
+    """
+
+    def __init__(self):
+        self.start_time = None
+        self.end_time = None
+
+    def __enter__(self):
+        self.start_time = time.time()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.end_time = time.time()
+        return False
+
+    def get_denials(self):
+        # Get AVC denials that occurred during the context manager period.
+        return _check_denials_with_ausearch(self.start_time, self.end_time)
+
+    def get_process_contexts(self):
+        # Get process contexts from execve events during the context manager period.
+        return _check_process_contexts_from_audit(self.start_time, self.end_time)
+
+
+@pytest.mark.tier2
+def test_register_unconfined_t_no_context_change(insights_client):
+    """
+    :id: 91c9c37f-b954-4fc4-84da-a4a2a9dbee9d
+    :title: Test insights-client --register runs in unconfined_t without context change
+    :description:
+        This test verifies that when running insights-client --register as root
+        in unconfined_t SELinux context with SELinux enforcing mode, the insights-core
+        code executes in unconfined_t context without any SELinux context change
+        and without any SELinux denials.
+    :tags: Tier 1
+    :reference: https://issues.redhat.com/browse/CCT-1717
+    :steps:
+        1. Register system using subscription-manager (already done by fixture)
+        2. Set SELinux to enforcing mode (using context manager)
+        3. Verify current context is unconfined_t
+        4. Run insights-client --register as root in unconfined_t context
+        5. Verify registration succeeded
+        6. Check audit logs for process execution contexts (SYSCALL execve events)
+        7. Check audit logs for SELinux AVC denials
+    :expectedresults:
+        1. System is registered (handled by register_subman fixture)
+        2. SELinux is set to enforcing mode and will be restored automatically
+        3. Current context is confirmed as unconfined_t
+        4. insights-client --register completes successfully
+        5. Registration status confirms system is registered
+        6. All insights-client and insights-core processes executed in
+           unconfined_t context
+        7. No SELinux AVC denials found in audit logs
+    """
+    if _get_selinux_mode() == "Disabled":
+        pytest.skip(reason="SELinux is disabled on this system")
+
+    with _selinux_mode("Enforcing"):
+        context = _get_current_context()
+        assert "unconfined_t" in context, f"Expected unconfined_t, got: {context}"
+
+        with SELinuxDenialsChecker() as checker:
+            result = insights_client.run("--register", selinux_context=context)
```

Comment:
The context does not have to be passed to achieve the same context for the new process. In that case, the `selinux_context` argument can be set just to `None`